frontend/src/metabase-lib/lib/metadata/Base.ts

```diff
@@ -1,14 +1,9 @@
-// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-// @ts-nocheck
 export default class Base {
-  _plainObject = null;
+  _plainObject: Record<string, unknown>;
 
   constructor(object = {}) {
     this._plainObject = object;
-
-    for (const property in object) {
-      this[property] = object[property];
```

Comment:
This line caused a type error that I couldn't figure out a way to get around, so replacing it with `Object.assign(this, object)` is more a little hack than anything else.

I had an idea to do something like `class Foo extends Base<{ a: number, ... }> { ... }` to avoid needing to declare props on `Foo`, but I couldn't figure out a way to do it -- not sure if it is possible to infer types on classes like this.